doc/sphinx/language/core/assumptions.rst

```diff
@@ -38,6 +38,11 @@ variable can be introduced at the same time. It is also possible to give
 the type of the variable as follows:
 :n:`(@ident : @type := @term)`.
 
+`(x : T | P)` is syntactic sugar for `(x : Stdlib.Init.Specif.sig (fun x : T => P))`
+i.e. `(x : {x : T | P})` (it is implemented using the first, so changing the
+`{x : T | P}` notation will not change the meaning of `(x : T | P)`,
+and changing the implicit arguments of `sig` will break `(x : T | P)`).
```

Comment:
You're making this same change in two PR?

Reply:
That's why the other one says it depends on this one